torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -445,20 +445,22 @@ void Scope::clear() {
 ForLoop::ForLoop(
     Passkey passkey,
     Val* index,
-    Val* extent,
     IterDomain* iter_domain,
+    Val* extent,
     bool unroll,
     bool vectorize,
     Val* shift)
     : Expr(passkey),
       index_{index},
-      extent_{extent},
       iter_domain_{iter_domain},
+      // Set extent_ as iter_domain's extent if parameter extent is null
+      extent_{extent != nullptr ? extent : iter_domain->extent()},
```

Comment:
why not just do this as the implementation of `extent()`? I don't imagine we would want mutable for loops, but seems pretty reasonable to just put this in the accessor.